packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseSchemaObject-test.js

```diff
@@ -342,5 +342,58 @@ describe('Schema Object', () => {
         expect(name.element).to.equal('name');
       });
     });
+
+    describe('#required', () => {
+      it('warns when required is not an array', () => {
+        const schema = new namespace.elements.Object({
+          type: 'object',
+          required: {},
+        });
+        const result = parse(context, schema);
+
+        expect(result.length).to.equal(2);
+
+        expect(result).to.contain.warning(
+          "'Schema Object' 'required' is not an array"
+        );
+      });
+
+      it('warns when required is not an array of strings', () => {
+        const schema = new namespace.elements.Object({
+          type: 'object',
+          required: [1, true],
+        });
+        const result = parse(context, schema);
+
+        expect(result.length).to.equal(3);
+
+        expect(result.warnings.toValue()).to.deep.equal([
+          "'Schema Object' 'required' is not an array of strings",
+          "'Schema Object' 'required' is not an array of strings",
```

Comment:
The warning refers to the whole array, but is created for every array item, which feels/reads strange. Either it should be there only once for the whole array, or we could have warnings like...

Okay, I couldn't come up with a better warning 😅 Having sth like `item #2 of 'required'` feels overly detailed and doesn't read so clearly. At the same time, having just one warning for the whole array is an unnecessary loss of source map to each of the issues. If the `required` looks like this:

```
required:
  - yes
  - no
  - true
```

each warning will have a different source map with a different line, even though they have the same text. So I guess what you have here is the best compromise solution after all.

Reply:
I have an idea, I'll update PR when I rebase it with new message

Reply:
Updated to be: `'Schema Object' 'required' array value is not a string`. Note in cases like the following:

```
required:
  - name
  - id
  - null
```

You'd get a single warning about the last value only with specific source map.